lib/config.cpp

```diff
@@ -363,9 +370,28 @@ json Config::load_interface_file(const std::string& intf_name) {
 
         json interface_json = json::parse(intf_file);
 
+        // this subschema can not use allOf with the draft-07 schema because that will cause our validator to
+        // add all draft-07 default values which never validate (the {"not": true} default contradicts everything)
+        // --> validating against draft-07 will be done in an extra step below
```

Comment:
It refers to a (slightly surprising) behavior of the validator that would populate the patch with these draft-7 default values. Which in turn would lead to some very wrong non-validatable output  that we encountered during testing. Might be useful for future reference, but maybe we should clarify the intent a bit more.

Reply:
which default values, if I have a non-required integer property and will validate against draft7 with patch, will this lead to the property been set to zero then, or what?

Reply:
I have an example from the **enable command** of the **board_support_AC** interface which we found during testing:
In the main branch the validator would create the following output:
```json
{"description":"True: enabled, false: disabled.","type":"boolean"}
```

Without the changes mentioned in the comment we encountered the following output after validating with draft-7 and using the patch from the validation.
```json
{"additionalItems":true,"additionalProperties":true,"contains":true,"description":"True: enabled, false: disabled.","else":true,"if":true,"items":true,"not":true,"propertyNames":true,"readOnly":false,"then":true,"type":"boolean","uniqueItems":false}
```
The way it's implemented now prevents these (obviously wrong) additions to be patched in.

Reply:
What kai said. Even worse: these additions are contradicting in itself: you never will be able to validate something having `{"not": true}` in the schema no matter what you try to validate.